test/e2e/network/service.go

```diff
@@ -4180,14 +4160,33 @@ func createPodOrFail(ctx context.Context, f *framework.Framework, ns, name strin
 }
 
 // launchHostExecPod launches a hostexec pod in the given namespace and waits
-// until it's Running
-func launchHostExecPod(ctx context.Context, client clientset.Interface, ns, name string) *v1.Pod {
+// until it's Running. If avoidNode is non-nil, it will ensure that the pod doesn't
+// land on that node.
+func launchHostExecPod(ctx context.Context, client clientset.Interface, ns, name string, avoidNode *string) *v1.Pod {
```

Comment:
should not be better to add  parameter that can mutate the pod generated?

```
func launchHostExecPod(ctx context.Context, client clientset.Interface, ns, name string, update func(*v1 Pod))
```

and we can do
```
execPod := launchHostExecPod(ctx, cs, namespace, "hostexec", func(pod *v1.Pod) {
			pod.Spec.Affinity = &v1.Affinity{
			NodeAffinity: &v1.NodeAffinity{
				RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
					NodeSelectorTerms: []v1.NodeSelectorTerm{{
						MatchFields: []v1.NodeSelectorRequirement{{
							Key:      "metadata.name",
							Operator: "NotIn",
							Values:   []string{*avoidNode},
						}},
					}},
				},
			},
		}
		}))
```

Reply:
yeah, we'd have to do the same thing in two places though. I just figured, since two of the callers need the same tweak, just add support for that tweak

Reply:
as you wish, is just that I prefer to avoid future us adding a new field to `launchHostExecPod` for any reasons if we can just do it once